src/Config/ConfigurationLoader.php

```diff
@@ -0,0 +1,116 @@
+        private readonly APIRequestWrapper $apiRequestWrapper,
+        private readonly IConfigurationStore $configurationStore,
+        private readonly int $cacheAgeLimit = 30
+    ) {
+        $this->parser = new UFCParser();
+    }
+
+    /**
+     * @throws InvalidApiKeyException
+     * @throws HttpRequestException
+     * @throws InvalidConfigurationException
+     */
+    public function getFlag(string $key): ?Flag
+    {
+        $this->reloadConfigurationIfExpired();
+        return $this->configurationStore->getFlag($key);
+    }
+
+    /**
+     * @param string $flagKey
+     * @param string $variation
+     * @return string|null
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidConfigurationException
+     */
+    public function getBanditByVariation(string $flagKey, string $variation): ?string
+    {
+        $this->reloadConfigurationIfExpired();
+        return $this->configurationStore->getBanditVariations()->getBanditByVariation($flagKey, $variation);
+    }
+
+    /**
+     * @param string $flagKey
+     * @return bool
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidConfigurationException
+     */
+    public function isBanditFlag(string $flagKey): bool
+    {
+        $this->reloadConfigurationIfExpired();
+        return $this->configurationStore->getBanditVariations()->isBanditFlag($flagKey);
+    }
+
+    /**
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidConfigurationException
+     */
+    public function reloadConfigurationIfExpired(): void
+    {
+        $cacheAge = $this->configurationStore->getFlagCacheAgeSeconds();
+        if ($cacheAge < 0 || $cacheAge >= $this->cacheAgeLimit) {
+            $this->fetchAndStoreConfigurations();
+        }
+    }
+
+    /**
+     * @throws HttpRequestException
+     * @throws InvalidApiKeyException
+     * @throws InvalidConfigurationException
+     */
+    public function fetchAndStoreConfigurations(): void
+    {
+        $responseData = json_decode($this->apiRequestWrapper->getUFC(), true);
+        if (!$responseData) {
+            syslog(LOG_WARNING, "[Eppo SDK] Empty or invalid response from the configuration server.");
+            return;
+        }
+
+        $inflated = array_map(fn($object) => $this->parser->parseFlag($object), $responseData['flags']);
+        $variations = [];
+        if (isset($responseData['bandits'])) {
+            $variations = array_map(
+                fn($listOfVariations) => array_map(fn($json) => BanditVariation::fromJson($json), $listOfVariations),
+                $responseData['bandits']
+            );
+        } else {
+            syslog(LOG_WARNING, "[EPPO SDK] No bandit-flag variations found in UFC response.");
+        }
+
+        $indexer = new BanditVariationIndexer($variations);
+
+        $banditModelResponse = json_decode($this->apiRequestWrapper->getBandits(), true);
```

Comment:
Actual change is here. File was moved in a pending PR so I think github is struggling a bit with the diffs on stacked PRs

Reply:
We only want to fetch bandits if the UFC says they are at play. Thus I'm thinking this should be rolled into within the `if (isset($responseData['bandits'])) {` conditional branch